tasks.py

```diff
@@ -74,39 +106,145 @@ def start(ctx, ports=SERVER_POOL_PORT, image=SRC_IMAGE_NAME, unsafe=False,
             {image} {command}".format(**cmd_params)
-        )
+        run_as_cmd = run_as(OS_NAME)
+        cmd = get_cmd(run_as_cmd, base_cmd)
+        ctx.run(cmd)
 
         while not os.path.isfile(CHECK_FILE_PATH):
             print("** Checking code server status. Press Ctrl-C to exit **\r", end="")
-        print("** Code server is up and running successfully **")
+        print("\n** Code server is up and running successfully **")
 
 
 @task
 def stop(ctx, container=TARGET_CONTAINER_NAME, hide=True):
-    result = ctx.run("sudo docker ps -q --filter='name={0}'".format(container))
+    base_filter_cmd = "docker ps -q --filter='name={0}'".format(container)
+    run_as_cmd = run_as(OS_NAME)
+    cmd = get_cmd(run_as_cmd, base_filter_cmd)
+    result = ctx.run(cmd)
+
     remove_check_file(CHECK_FILE_PATH)
     if result.stdout:
-        print ("** Discarding the docker container <{0}>".format(container))
-        ctx.run("sudo docker stop {0}".format(container))
-        ctx.run("sudo docker rm {0}".format(container))
-        print ("** Done! Discarded the docker container <{0}>".format(container))
+        print ("** Stopping the docker container <{0}> **".format(container))
+        base_stop_cmd = "docker stop {0}".format(container)
+        cmd = get_cmd(run_as_cmd, base_stop_cmd)
+        ctx.run(cmd)
+        print ("** Done! Stopped the docker container <{0}> **".format(container))
+
+        print ("** Discarding the docker container <{0}> **".format(container))
+        base_rm_cmd = "docker rm {0}".format(container)
+        cmd = get_cmd(run_as_cmd, base_rm_cmd)
+        ctx.run(cmd)
+        print ("** Done! Discarded the docker container <{0}> **".format(container))
     else:
         print("** Docker container <{0}> not found **".format(container))
+
+# Docker compose based deployment
+@task
+def build(ctx):
+    run_as_cmd = run_as(OS_NAME)
+
+    copy_tree(
+        os.path.join(SCRIPT_DIR, 'requirements'),
+        os.path.join(SCRIPT_DIR, 'docker', 'Files')
+    )
+
+    base_build_cmd = "docker-compose build --no-cache"
+    cmd = get_cmd(run_as_cmd, base_build_cmd)
+    print ("** Building docker images **")
+    ctx.run(cmd)
+    print ("** Done! Built the docker images **")
+
+    base_build_cmd = "docker pull mariadb:10.2 "
+    cmd = get_cmd(run_as_cmd, base_build_cmd)
+    print ("** Pulling maria-db base image **")
+    ctx.run(cmd)
+    print ("** Done! Pulled maria-db base image **")
+
+@task
+def begin(ctx):
+    print("** Initializing docker containers **")
+    base_cmd = "docker-compose up -d"
+    run_as_cmd = run_as(OS_NAME)
+    cmd = get_cmd(run_as_cmd, base_cmd)
+    ctx.run(cmd)
+    print ("** Done! Initialized the docker containers **")
+
+@task(begin)
+def deploy(ctx, fixtures=False):
+    run_as_cmd = run_as(OS_NAME)
+
+    print("** Setting up & migrating database **")
+    base_make_migrate_cmd = "docker exec -i yaksh_django" \
+        " python3 manage.py makemigrations"
```

Comment:
This can't be python3 all the time. It should be python 2 or 3

Reply:
Dockerfile installation is done for Python3, hence for deployment purposes we stick to Python 3